### PR TITLE
Added Receipt Yaml files and changes in tests, yaml file changes for …

### PR DESCRIPTION
**Pull request by manojsalunke85**
…postmsg tests

Signed-off-by: manoj <manoj.salunke@intel.com>

### CODE REVIEW

File: tests/automation_framework/data/receipt/work_order_create_receipt.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    requesterGeneratedNonce: ''
+    requesterId: '0x3456'
+    requesterSignature: ''
+    signatureRules: ''
+    workOrderId: ''
+    workOrderRequestHash: ''
+    workerId: ''
+    workerServiceId: ''
+test_create_work_order_receipt_hexstr_workorderRequesthash:
+  workOrderRequestHash: '"98D0189C0DCA55496BBA7A932FBB870CA1F481A3D2EB5542E484BD89329D4CA6"'
```
Comment (anjalirai-intel) [inline tests/automation_framework/data/receipt/work_order_create_receipt.yaml]: Why we have to specify string inside single quotes and double quotes

File: tests/automation_framework/data/work_order/work_order_submit.yaml [hunk 2]
```diff
@@ -1139,19 +1139,76 @@ test_workordersubmit_workloadId_twoworkload:
...
 test_workordersubmit_workorderid_increasedhexlength:
-  workOrderId: '"0x9b48513c40fda73f0x9b48513c40fda73f0x9b48513c40fda73f"'
+  workOrderId: '0x9b48513c40fda73f0x9b48513c40fda73f0x9b48513c40fda73f'
 test_workordersubmit_workorderidworkloadid_same:
-  workOrderId: '"0xABCDY"'
-  workloadId: '"0xABCDY"'
+  workOrderId: '0xABCDY'
+  workloadId: '0xABCDY'
+test_workordersubmit_params_unknownparameter:
+  params: '{"unknownEncoding":"unknownEncoding"}'
```
Comment (anjalirai-intel) [inline tests/automation_framework/data/work_order/work_order_submit.yaml]: Above comment applicable

File: tests/automation_framework/data/work_order/work_order_submit.yaml [hunk 3]
```diff
@@ -1139,19 +1139,76 @@ test_workordersubmit_workloadId_twoworkload:
...
+test_workordersubmit_payloadFormat_notJSONRPC:
+  params: '{"payloadFormat": "NOT_JSON_RPC"}'
+test_workordersubmit_params_empty:
```
Comment (anjalirai-intel) [inline tests/automation_framework/data/work_order/work_order_submit.yaml]: params : {}, should also work

File: tests/automation_framework/tests/work_order_tests/test_submit.py [hunk 4]
```diff
@@ -913,7 +905,7 @@ def test_workordersubmit_workerId_notdefaultlength(self):
                 result_response,
                 "worker "
                 "6ba1f459476bc43b65fd554f6b65910a8f551e4bcb"
-                "0eee6a96dcebaeb14f2ae923456234564567 "
+                "0eee6a96dcebaeb14f2ae923456234564567"
```
Comment (anjalirai-intel) [inline tests/automation_framework/tests/work_order_tests/test_submit.py]: In the end of this string space should be there

File: tests/automation_framework/src/libs/test_base.py [hunk 5]
```diff
@@ -242,58 +243,97 @@ def post_json_msg(self, request_file):
         logger.info('**********Received Response*********\n%s\n', response)
         return response
 
-    def run_test(self, config_file):
-        test_name = inspect.stack()[1].function
-        test_data = read_config(config_file, test_name)
-        result_response = ""
+    def avalon_uncomputed(self, request_file=None, test_data=None):
```
Comment (jinengandhi-intel) [inline tests/automation_framework/src/libs/test_base.py]: Do you need this request_file parameter and the else part as we would now always send the test_data? I feel the function should just have the json.dumps and then the postmsg

**Review by jinengandhi-intel — CHANGES_REQUESTED** [review]

File: tests/automation_framework/src/utilities/worker_utilities.py [hunk 6]
```diff
@@ -245,7 +247,7 @@ def config_data_update(input, key, value):
 
 def read_config(config_file, test_name):
```
Comment (jinengandhi-intel) [inline tests/automation_framework/src/utilities/worker_utilities.py:233]: Hope the modifications to the yaml file reading has been tested. Rest it looks okay.

**Review by jinengandhi-intel — APPROVED** [review]

File: tests/automation_framework/src/utilities/worker_utilities.py (hunk 6 above)
Comment (manojsalunke85) [reply tests/automation_framework/src/utilities/worker_utilities.py:233]: Yes. Whole test suite is run with sdk and listener.

File: tests/automation_framework/data/work_order/work_order_submit.yaml [hunk 7]
```diff
@@ -32,93 +33,53 @@ Default:
...
-    "dataHash": "",
-
-    "data": "perl",
-
-    "encryptedDataEncryptionKey": "",
-
-    "iv": ""
-
-    }]'
+  '"inData"':
```
Comment (anjalirai-intel) [inline tests/automation_framework/data/work_order/work_order_submit.yaml]: This keyword is still under single quotes and double quotes

File: tests/automation_framework/data/work_order/work_order_submit.yaml [hunk 8]
```diff
@@ -177,903 +120,425 @@ test_workordersubmit_indata_bothindexremoveDataDatahash:
...
-    }
-
-    ]}'
+  params:
+    outData:
+    - data: ''
+      encryptedDataEncryptionKey: ''
+      index: 0
+      iv: ''
+test_workordersubmit_params_empty:
```
Comment (anjalirai-intel) [inline tests/automation_framework/data/work_order/work_order_submit.yaml:508]: For this tests "params :{}" should also work

File: tests/automation_framework/data/work_order/work_order_submit.yaml [hunk 9]
```diff
@@ -1083,75 +548,87 @@ test_workordersubmit_requestersignature_differentlength:
...
   sessionKeyIv: \\@!$&*
 test_workordersubmit_workerEncryptionKey_notdefaulthex:
-  workerEncryptionKey: '234021262340212623402126'
+  workerEncryptionKey: 234021262340212623402126
 test_workordersubmit_workerEncryptionKey_special_character:
   workerEncryptionKey: hs$vi
 test_workordersubmit_workerId_notdefaultlength:
   workerId: 6ba1f459476bc43b65fd554f6b65910a8f551e4bcb0eee6a96dcebaeb14f2ae923456234564567
+test_workordersubmit_workerId_notdefaultlength_postmsg:
+  '"inData"':
```
Comment (anjalirai-intel) [inline tests/automation_framework/data/work_order/work_order_submit.yaml]: This keyword is still under single quotes and double quotes

File: tests/automation_framework/src/utilities/worker_utilities.py [hunk 10]
```diff
@@ -212,28 +212,30 @@ def configure_data(action_obj, input_json, worker_obj, pre_test_response):
 def handle_value(val):
```
Comment (anjalirai-intel) [inline tests/automation_framework/src/utilities/worker_utilities.py]: The function is not being used. Delete it

File: tests/automation_framework/data/work_order/work_order_submit.yaml (hunk 8 above)
Comment (manojsalunke85) [reply tests/automation_framework/data/work_order/work_order_submit.yaml:508]: I just tried it. It didn't work. still took all the default values. Keeping it the same for now.

File: tests/automation_framework/data/work_order/work_order_submit.yaml (hunk 7 above)
Comment (manojsalunke85) [reply tests/automation_framework/data/work_order/work_order_submit.yaml]: Fixed

File: tests/automation_framework/src/libs/test_base.py (hunk 5 above)
Comment (manojsalunke85) [reply tests/automation_framework/src/libs/test_base.py]: fixed

File: tests/automation_framework/data/receipt/work_order_create_receipt.yaml (hunk 1 above)
Comment (manojsalunke85) [reply tests/automation_framework/data/receipt/work_order_create_receipt.yaml]: fixed

File: tests/automation_framework/data/work_order/work_order_submit.yaml (hunk 2 above)
Comment (manojsalunke85) [reply tests/automation_framework/data/work_order/work_order_submit.yaml]: fixed

File: tests/automation_framework/data/work_order/work_order_submit.yaml (hunk 3 above)
Comment (manojsalunke85) [reply tests/automation_framework/data/work_order/work_order_submit.yaml]: Keeping it the same for now. Iit didnt work with params: {}

File: tests/automation_framework/tests/work_order_tests/test_submit.py (hunk 4 above)
Comment (manojsalunke85) [reply tests/automation_framework/tests/work_order_tests/test_submit.py]: Fixed

File: tests/automation_framework/data/work_order/work_order_submit.yaml (hunk 9 above)
Comment (manojsalunke85) [reply tests/automation_framework/data/work_order/work_order_submit.yaml]: Fixed

File: tests/automation_framework/src/utilities/worker_utilities.py (hunk 10 above)
Comment (manojsalunke85) [reply tests/automation_framework/src/utilities/worker_utilities.py]: Fixed

**Review by anjalirai-intel — APPROVED** [review]

**Pull request merged**